### DISCUSSION TITLE
How to get color kitty icon in titlebar with Wayland?

### DISCUSSION THREAD

**Discussion by egnor** [kovidgoyal/kitty#3833] [Q&A] [answered]
When I run kitty under Wayland, the client side decorations look like this:

![image](https://user-images.githubusercontent.com/279819/125890625-82734dcf-7de7-40d2-8b70-3161b0f30487.png)

That's OK but the cat face could be better. In a [tangential thread about Wayland decorations](https://github.com/kovidgoyal/kitty/issues/3284#issuecomment-826250911), @stellarexplosions posted a screenshot:

![image](https://user-images.githubusercontent.com/11005389/115980404-19c07a00-a5aa-11eb-93d6-8989d6f82375.png)

That's an adorable cat! And so nicely centered in the title bar! How do I get that? I have the noto-color-emoji font installed, and my `.fonts.conf` set up to prepend it -- in fact, you'll note that kitty's own unicode picker gets perfectly happy colored kitties:

![image](https://user-images.githubusercontent.com/279819/125890969-e83561ef-aeaa-4d93-9be2-dc1ce30ee016.png)

I just don't get one in the title bar. :crying_cat_face: What am I missing?

-- Kitty lover (both kinds) :heart_eyes_cat:

**Comment by kovidgoyal** [comment]
No clue, works for me. The relevant code is in find_fallback_font_for()
in freetype_render_ui_text.c

**Reply by egnor** [reply]
When I run [the test from the comments](https://github.com/kovidgoyal/kitty/blob/master/kitty/freetype_render_ui_text.c#L497) in `render_line()`

```
kitty +runpy "from kitty.fast_data_types import *; open('/tmp/test.rgba', 'wb').write(freetype_render_line())" &&
convert -size 800x60 -depth 8 /tmp/test.rgba /tmp/test.png &&
icat /tmp/test.png
```

I get this image, which does render the rocket in color, but renders the cat in B&W:
![test](https://user-images.githubusercontent.com/279819/125912078-6d8f315e-6076-4fb5-844a-ccfeb9ee8916.png)

I presume you get a full-color cat when you do that?

**Comment by kovidgoyal** [comment, marked as answer]
On Fri, Jul 16, 2021 at 12:54:59AM -0700, Daniel Egnor wrote:
> I presume you get a full-color cat when you do that?

Yes, I do. Probably just add a rule in fontconfig for that character to
use noto

**Reply by egnor** [reply]
I can do better than that!

The issue seems to be as described in https://github.com/stove-panini/fontconfig-emoji, since the title bar specifies no family whatsoever, on this system anyway (Ubuntu 21.04) it falls back to DejaVu Sans, which includes some (ugly) emoji glyphs. Since this becomes the "declared" family, DejaVu Sans takes precedence over the color emoji font, and DejaVu's (ugly) emoji glyphs get used instead of the pretty color emoji.

Presumably on your system there's a somewhat different setup for some reason.

The emoji in the actual terminal work fine, because Kitty selects a specific font, which (like most non-DejaVu specific fonts) does not include emoji glyphs, so the stacking order of (selected font),(emoji font),(fallback fonts) works fine.

The advice in the repo linked above is to blacklist DejaVu fonts. Not sure if that's the best advice, but it certainly does fix the problem with the title bar kitty cat.

This is clearly a system problem, not a Kitty issue. Also, perhaps at some point you will start actively picking the font used for the client side decoration, rather than just taking the absolute system default font. (Though really the whole CSD concept is strange and should probably be thought of as a fallback stopgap, not something to make pretty? It seriously doesn't make sense for every app to invent its own title bars.)

**Comment by kovidgoyal** [comment]
On Fri, Jul 16, 2021 at 01:30:59AM -0700, Daniel Egnor wrote:
> I can do better than that!
> 
> The issue seems to be as described in https://github.com/stove-panini/fontconfig-emoji, since the title bar specifies no family whatsoever, on this system anyway (Ubuntu 21.04) it falls back to DejaVu Sans, which includes some (ugly) emoji glyphs. Since this becomes the "declared" family, it takes precedence over the color emoji font, and its (ugly) emoji glyphs get used.
> 
> Presumably on your system there's a somewhat different setup for some reason.

Yes, I have fontconfig configured with default fonts that I like. 

> 
> The advice in the repo linked above is to blacklist DejaVu fonts. Not sure if that's the best advice, but it certainly does fix the problem with the title bar kitty cat.

DejaVu is a pretty bad font generally, it has metrics issues with some
characters for instance (search for dejavu in this issue tracker).
I endorse un-installing/blacklisting it.

> 
> This is clearly a system problem. Also, perhaps at some point you will start actively picking the font used for the client side decoration, rather than just taking the absolute system default font. (Though really the whole CSD concept is strange and should probably be thought of as a fallback stopgap, not something to make pretty? It seriously doesn't make sense for every app to invent its own title bars.)

I have no plans to specify a font family for this, ideally systems
should be configured with whatever font they like. Sadly, this being
linux, there is no central place to configure fonts, every desktop
env/application invents its own, despite the existence of fontconfig.

Personally, I just configure it once in fontconfig, and forget about it.
Most applications fallback to those.
